src/datachain/lib/file.py

```diff
@@ -432,14 +354,32 @@ class ImageFile(File):
 
     def read(self):
         """Returns `PIL.Image.Image` object."""
-        fobj = super().read()
-        return Image.open(BytesIO(fobj))
+        with self.open() as stream:
+            return Image.open(BytesIO(stream.read()))
 
     def save(self, destination: str):
         """Writes it's content to destination"""
         self.read().save(destination)
 
 
+class TarVFile(File):
+    """`DataModel` for files extracted from tar archives."""
+
+    tar: File
+
+    @contextmanager
+    def open(self):
+        """Stream file from tar archive."""
+        vpath = str(PurePosixPath(self.path).relative_to(self.tar.path))
```

Comment:
just to make sure - I assume this should be fine with cache and things like `get_local_path` (?)

Reply:
Since `self.path` is just a pydantic str field, it should be fine here. Not sure `get_local_path` could be impacted in any way unless I'm missing something.